python/tvm/testing/aot.py

```diff
@@ -415,24 +415,23 @@ def _emit_main_compare(main_file, outputs, output_tolerance, mod_name, use_inter
             comparison_function = "fabs"
             tolerance = output_tolerance or 0.001
 
-        data_length_var_name = (
-            _mangle_name(mod_name, f"output_data_{sanitized_tensor_name}") + "_len"
+        actual_data_name = _mangle_name(mod_name, f"output_data_{sanitized_tensor_name}")
+        data_len_var = actual_data_name + "_len"
+        main_file.write(
+            f"const size_t {data_len_var}"
+            f"= sizeof({actual_data_name})/sizeof({actual_data_name}[0]);\n"
```

Comment:
We generate `actual_data_name` and `data_length_var_name` in the inputs/outputs already, why are we recalculating it here based on types we've defined within the AOT test harness?

We probably need to add a `_LEN` macro to the MLF header?

Reply:
`_LENGTH` to be clearer 😸

Reply:
As discussed offline, the two variables `LEN` and `SIZE` serve different purposes. `SIZE` will come from MLF export whereas `LEN` will be part of the AOT test harness.